### PR TITLE
No longer use root logger

### PR DESCRIPTION
**Pull request by philippjfr**
Supersedes https://github.com/pyviz/param/pull/308

### CODE REVIEW

File: param/parameterized.py [hunk 1]
```diff
@@ -31,17 +31,27 @@
...
+# logger named `"param"` will be instantiated. If `name` is set, a descendant
+# logger with the name ``"param.<name>"`` is returned (or
+# ``logger.name + ".<name>"``)
 logger = None
-def get_logger():
+def get_logger(name=None):
     if logger is None:
-        # If it was not configured before, do default initialization
-        if not logging.getLogger().handlers:
-            logging.basicConfig(level=INFO)
```
Comment (philippjfr) [inline param/parameterized.py]: Should we keep this?

File: param/parameterized.py [hunk 2]
```diff
@@ -1684,14 +1693,12 @@ def __db_print(self_,level,msg,*args,**kw):
...
-        if get_logger().isEnabledFor(level):
+        if get_logger(name=self_or_cls.name).isEnabledFor(level):
 
             if dbprint_prefix and callable(dbprint_prefix):
-                prefix=dbprint_prefix() # pylint: disable-msg=E1102
-            else:
-                prefix=""
+                msg = dbprint_prefix() + ": " + msg  # pylint: disable-msg=E1102
 
-            get_logger().log(level, '%s%s: '+msg, prefix, self_or_cls.name, *args, **kw)
```
Comment (jbednar) [inline param/parameterized.py]: Is this going to change the formatting of log messages?

File: param/parameterized.py (hunk 1 above)
Comment (jbednar) [reply param/parameterized.py]: Now that we're making our own logger, I think we should explicitly set the default, yes.

File: param/parameterized.py (hunk 2 above)
Comment (philippjfr) [reply param/parameterized.py]: Yes, a warning now looks like this:

>WARNING:param.Curve01322: Some warning

It used to look like this:

>WARNING:root:Curve01315: Some warning

I'd prefer it if it was something like:

>WARNING:param:Curve01315: Some warning

but the ``.`` separator is the canonical way to define logger descendants.

Comment (jbednar) [reply param/parameterized.py]: Ok.  Sounds fine, and I agree with your preference, but it's not too bad.

**Comment by jbednar** [comment]
@jlstevens , I think this is mergeable, but if you have any objections, please make that clear asap!  

I think the changes to the warning messages mean that this release should be 1.9 and not 1.8.3, because even though it's not strictly part of the API people may still be relying on the warning message format using scripts that grep for warnings or errors, so it's important to acknowledge the change.

**Comment by philippjfr** [comment]
>I think the changes to the warning messages mean that this release should be 1.9 and not 1.8.3

This was already the plan.

**Comment by jlstevens** [comment]
I think this is a good idea and I'm happy to merge now. If this PR results in some unexpected behavior somewhere, I think that is more acceptable than modifying the root logger which can cause all sorts of problems. Merging.

**Pull request merged**